### DISCUSSION TITLE
Unable to point to specific Git branch

### DISCUSSION THREAD

**Discussion by ewash1** [fluxcd/flux2#2498] [Q&A] [answered]
Hey all,

I recently upgraded flux from 0.24.1 to 0.27.2 and noticed that I'm unable to point my GitRepository resource to a specific branch. I utilized this feature pretty heavily to test changes. Basically as soon as I edit the GitRepository to point to a specific branch, it appears to revert back to master almost immediately. 

![image](https://user-images.githubusercontent.com/94005123/156629893-8782e182-8b6a-4be2-a3f2-e5f6ab00d572.png)

Checking logs from the kustomize-controller show:

![image](https://user-images.githubusercontent.com/94005123/156633327-fd6f985f-af60-4716-9ea6-d6d68269dd5f.png)

Looking for thoughts on how to fix this or an explanation on what's going on here! Thanks!

**Comment by stefanprodan** [comment, marked as answer]
> Basically as soon as I edit the GitRepository to point to a specific branch, it appears to revert back to master almost immediately.

Is this edit done in cluster with kubectl?

**Reply by ewash1** [reply]
yes, using kubectl edit.

**Reply by stefanprodan** [reply]
See here how to use kubectl with Flux: https://fluxcd.io/docs/faq/#why-are-kubectl-edits-rolled-back-by-flux

**Reply by ewash1** [reply]
thanks for the help Stefan, we're able to point to specific branches now.
